lms/static/scripts/frontend_apps/utils/youtube.ts

```diff
@@ -34,33 +34,35 @@ export function videoIdFromYouTubeURL(url: string): string | null {
   return match?.[2] ?? null;
 }
 
+type YouTubeVideoInfo = {
+  isLoading: boolean;
+  image?: string;
+  title?: string;
+  channel?: string;
+  error?: 'video_not_found' | 'unknown';
+};
+
 /* istanbul ignore next */
-export function useYouTubeVideoInfo(videoId: string | null) {
-  // TODO Use dummy data for now, until a proper API has been implemented
-  const [thumbnail, setThumbnail] = useState<ThumbnailData>();
-  const [metadata, setMetadata] = useState<{
-    title: string;
-    channel: string;
-  }>();
+export function useYouTubeVideoInfo(videoId: string | null): YouTubeVideoInfo {
```

Comment:
I'm not sure this hook adds a lot of value compared to using `useAPIFetch` directly in the `YouTubePicker` component. We're not using the API call in multiple places, and it isn't really abstracting away any of the information from the component.

Reply:
I was trying to decouple a bit the component with the way the video info is fetched (it's an API call in this case, but that is not relevant for the component implementation).

The hook allows to "hide" the API call, and do a bit of response mapping, although it's true this API call returns the required info almost verbatim.

Something else I had in mind is that this could allow to mock the import of this hook and simplify the component test, while we then test the hook itself separately.

I'll remove it for now and we can get it back in the future if we see the need.